src/overlays/actors/ovl_Bg_F40_Switch/z_bg_f40_switch.c

```diff
@@ -34,37 +34,167 @@ const ActorInit Bg_F40_Switch_InitVars = {
     (ActorFunc)BgF40Switch_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80BC4E04[] = {
+s32 sBgF40SwitchGlobalsInitialized = false;
+u32 sBgF40SwitchLastUpdateFrame;
+
+extern CollisionHeader object_f40_switch_Colheader_000118;
+extern Gfx object_f40_switch_DL_000438;
+extern Gfx object_f40_switch_DL_000390;
+
+/*
+ * Updates all instances of this actor in the current room, unless it's already been called this frame.
+ */
+void func_80BC47B0(BgF40Switch* this, GlobalContext* globalCtx) {
+    if (globalCtx->gameplayFrames != sBgF40SwitchLastUpdateFrame) {
+        u32 pressedSwitchFlags[4] = { 0 };
+        u32 pad;
+        s32 switchFlag;
+        s32 isPressed;
+        Actor* actor;
+        BgF40Switch* actorAsSwitch;
+        u32 unk40;
+
+        unk40 = func_801233E4(globalCtx);
+        for (actor = globalCtx->actorCtx.actorList[ACTORCAT_SWITCH].first; actor != NULL; actor = actor->next) {
```

Comment:
You can put the variable assignation in the same line of the declaration if it matches.

```suggestion
        BgF40Switch* actorAsSwitch;
        u32 unk40 = func_801233E4(globalCtx);

        for (actor = globalCtx->actorCtx.actorList[ACTORCAT_SWITCH].first; actor != NULL; actor = actor->next) {
```

Reply:
Since this is not a trivial initialization but rather calls a function, I felt it was more appropriate to put it together with the code, not together with the declarations, if they are separated.